config/elasticache/config.go

```diff
@@ -164,6 +164,19 @@ func Configure(p *config.Provider) { //nolint:gocyclo
 		)
 	})
 
+	p.AddResourceConfigurator("aws_elasticache_serverless_cache", func(r *config.Resource) {
+		r.UseAsync = true
+		r.References["kms_key_id"] = config.Reference{
+			TerraformName: "aws_kms_key",
+		}
+		r.TerraformCustomDiff = func(diff *terraform.InstanceDiff, _ *terraform.InstanceState, _ *terraform.ResourceConfig) (*terraform.InstanceDiff, error) {
+			if diff != nil && diff.Attributes != nil {
+				delete(diff.Attributes, "security_group_names.#")
+			}
+			return diff, nil
+		}
```

Comment:
Why do we add this func?

Reply:
This was present in the replicationgroup resource. This seemed to have been added during a TF Update recently and I maybe "assumed" that serverlesscache would have the same error as we saw during TF Provider Update.

Reply:
I see; however, we usually avoid adding this kind of configuration unless they are really required. My suggestion is please remove it, test it, and add it if really required. What do you think?

Reply:
Since the [uptest](https://github.com/crossplane-contrib/provider-upjet-aws/actions/runs/10273838435) passed successfully, we can proceed without the `TerraformCustomDiff` function, thank you.